project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.module

```diff
@@ -298,6 +298,21 @@ function c4m_user_profile_user_login(&$edit, $user) {
 }
 
 /**
+ * Implements hook_mail_alter().
+ *
+ * For users that requested email notifications format to be plaintext,
+ * sets 'email confirmation' mail format accordingly.
+ */
+function c4m_user_profile_mail_alter(&$message) {
+  if ($message['id'] != 'email_confirm_confirmation_email') {
```

Comment:
Mime mail module takes care of it - `mimemail_prepare_message()` function.
It loads the $account (load by mail) and checks if it has palin text option enabled.
Confirmation of new mail is special case, because it's impossible to load user by new mail - it does not exist.
Therefore, I add special logic only for this case.